File: 404/sum-of-left-leaves.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function (root) {
    let ans = 0;

    root && root.left && dfs(root.left, 1);
    root && root.right && dfs(root.right, 0);

    function dfs(node, pos) {
        pos && node && !node.left && !node.right && (ans += node.val);

        node && node.left && dfs(node.left, 1);
        node && node.right && dfs(node.right, 0);
    }

    return ans;
};

var sumOfLeftLeaves = function(root) {
    function loop(root, isLeft) {
        if (!root) return 0;
        else if (!root.left && !root.right && isLeft) return root.val;
        else return loop(root.left, true) + loop(root.right, false);
    }
    return loop(root, false);
}

File: 404/sum-of-left-leaves.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func sumOfLeftLeaves(root *TreeNode) int {
	var loop func(*TreeNode, bool) int
	loop = func(root *TreeNode, isLeft bool) int {
		if root == nil {
			return 0
		} else if root.Left == nil && root.Right == nil && isLeft {
			return root.Val
		} else {
			return loop(root.Left, true) + loop(root.Right, false)
		}
	}

	return loop(root, false)
}
